Update Typescript to 3.8

cfcosta opened the pull request:
Depends on https://github.com/kiali/kiali-ui/pull/1654.

Fixes https://github.com/kiali/kiali/issues/2220.

Fixes all failing compilation errors, waiting for test runs.

cfcosta commented:
All tests are passed, this seems good to go. Let me get some QE involvement.

aljesusg commented on src/components/Nav/Masthead/Masthead.tsx:
```diff
@@ -12,7 +12,7 @@ class Masthead extends React.Component {
       <ToolbarNext>
         <ToolbarGroup>
-          <PfSpinner />
+          <PfSpinner isLoading={false} />
```
Why are you passing isLoading when this value is from redux ? 

```ts
const mapStateToProps = (state: KialiAppState) => ({
  isLoading: state.globalState.loadingCounter > 0
});
```

cfcosta replied:
Good catch, there's something funky going on with properties coming from redux (they can not be marked as required because they might not be there).

aljesusg commented on package.json:
```diff
@@ -129,7 +129,7 @@
     "source-map-explorer": "1.8.0",
-    "typescript": "3.5.3"
+    "typescript": "3.7.5"
   },
```
Why we need to upgrade this ? In https://github.com/kiali/kiali-ui/pull/1654 the test passed correctly

cfcosta replied:
Yeah, but this PR is exactly to upgrade Typescript :)

jshaughn commented:
> The way to present these PR is not right.

@lucasponce @cfcosta , so after some discussion hopefully we've fixed things process-wise.  Issue https://github.com/kiali/kiali-ui/pull/1561 is on the sprint backlog.   Originally for that issue there was an investigation of TS 3.7 as a potential helper for the solution.  In the end the 3.7 update was not required for the fix but in a reversal, the fix solves a compilation problem when using TS 3.7.  In the meantime @jotak created https://github.com/kiali/kiali/issues/2220 suggesting an update to TS 3.7.  So, because the TS 3.7 update was already done and dev-tested by @cfcosta , he generated this PR for the update.

There were a couple or process issues that we have tried to rectify:

- The PR description lacked the "Fixes kiali/kiali#2220" string.  It has now been added.
- PR review can not be added for a PR for an issue that is not on the sprint backlog (nor assigned)
  - I have removed the reviewers with the exception of those "Recommened" reviewers that github seems not to offer a removal option.
  - I have added the Discussion Needed label to 2220 to see whether it should be put on the backlog.
- No comment was added to explain this oddity of a PR being generated
   - I've added this long diatribe...

So, basically folks just need to remember the process:

Issue -> Discussion Needed -> Backlog -> Sprint Backlog -> Assigned -> In Progress -> PR(s) -> Review -> Merge.

If for some reason you have a PR it must manage to be for an issue and can not be reviewed until that issue is In Progress.  Just follow these steps and all should be right with the world.

jotak commented:
If the workflow process issue has been sorted out / clarified, is there any other reason to block this?

jshaughn commented:
@jotak I'm OK adding the issue to the Sprint 36 backlog as a maintenance task.  I'd prefer not to do it in Sprint 35 because this upcoming build may very well be included with Istio 1.5 and I'd like to avoid ths lib change late in the game.  So, let me know and I'll add it for the next sprint.

lucasponce commented:
Commented that this PR should wait the global CSS cleaning that @aljesusg is leading to avoid rebase conflicts.

Please, @aljesusg when you have completed the cleaning, if you can help @cfcosta to review this one and check that would need that's good.

mtho11 commented:
@cfcosta looks like this needs rebasing

mtho11 commented on package.json:
```diff
@@ -126,7 +126,7 @@
     "snyk": "^1.161.1",
     "source-map-explorer": "1.8.0",
-    "typescript": "3.5.3"
+    "typescript": "3.7.5"
```
Typescript is now at version: 3.8.3

cfcosta replied:
Ok, will do a proper testing again with the new version.

cfcosta commented:
So, after some trial and error everything seems ok. The build was failing because `incremental` can not be set together with `noEmit` on the tsconfig. This happens because we are using babel for the transpilation, so we get bundling and stuff like that.

I did some tests on master and on this branch, and as it turns out, `incremental` was not doing anything. The build times are the same with and without, unfortunately.

So I removed it and updated to 3.8.3,  @mtho11.

jshaughn commented:
Adding DNM label because I don't want a library change just before a sprint release.  We can merge on Monday assuming we get approval.

cfcosta commented:
@pbajjuri20 @mtho11 @aljesusg can you do a quick review? This is ready for merge.

pbajjuri20 commented:
Tested and passed on Kiali UI 1.15.0 (dc914d2da8337958e1db245b15567a28d31232dd)

jshaughn commented:
Removing DNM as 1.15 has been released.

mtho11 commented:
@lucasponce @aljesusg do you guys still have requested changes on this PR?
